source/binary.cpp

```diff
@@ -732,6 +739,43 @@ spv_result_t Parser::setNumericTypeInfoForType(
   return SPV_SUCCESS;
 }
 
+spv_result_t Parser::verifyLiteralUpperBits(
+    const spv_parsed_operand_t& parsed_operand, uint32_t word) {
+  // Verify that the upper bits are either set to 0, or sign-extended
+  auto verify = [&](uint32_t mask, std::vector<uint32_t> valid_patterns)
```

Comment:
FYI:  The style guide we're using has some non-obvious advice about lambda captures.  See https://google.github.io/styleguide/cppguide.html#Lambda_expressions  

"Use default capture by reference ([&]) only when the lifetime of the lambda is obviously shorter than any potential captures."

Your use here is ok by that guide.  :-)